examples/simple/io_write_dumpfile.py

```diff
@@ -8,12 +8,13 @@
 import random
 import sys
 from mitmproxy import io, http
+import typing  # noqa
 
 
 class Writer:
     def __init__(self, path: str) -> None:
         if path == "-":
-            f = sys.stdout  # type: io.TextIO
+            f = sys.stdout  # type: typing.IO[typing.Any]
```

Comment:
The FlowWriter only writes bytes, so this should really be `typing.IO[bytes]`. If that raises errors, that's a strong indicator that the code here does not work as intended. I'd be fine with removing the `if path == "-":` special case altogether.